extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -50,125 +43,38 @@ class AmpSocialShare extends AMP.BaseElement {
-      let paramValue = this.config_[param] || this.getDefaultValue_(
-          param, this.config_);
-      user.assert(!paramConf['required'] || paramValue !== null,
-          param + ' is a required attribute for ' + this.type_ + '. %s',
-        this.element);
-      if (paramValue == null && paramConf['type'] == 'fixed') {
-        paramValue = paramConf['value'];
-      }
-      if ('maxlength' in paramConf) {
-        const maxlength = paramConf['maxlength'];
-        user.assert(!paramValue || paramValue.length < maxlength,
-            param + ' cannot exceed ' + maxlength + '. %s', this.element);
-      }
-      if (paramValue != null) {
-        urlParams[paramConf['param']] = paramValue;
-      }
-    }
-
-    // Get the anchor or create one.
-    let link = elementByTag(this.element, 'a');
-    if (link == null) {
-      link = this.getWin().document.createElement('a');
-      link.textContent = this.typeConfig_['text'];
-      link.setAttribute('target', '_blank');
-
-      // Get the container or create one.
-      let container = elementByTag(this.element, 'span');
-      if (container == null) {
-        container = this.getWin().document.createElement('span');
-        container.classList.add(CLASSNAME_PREFIX + this.type_);
-
-        // Only add the container to the element if it didn't exist
-        this.element.appendChild(container);
-      }
-      container.appendChild(link);
-    }
-
-    // Set share url.
-    link.setAttribute('href',
-      addParamsToUrl(this.typeConfig_['url'], urlParams));
+    this.params_ = Object.assign({}, typeConfig.defaultParams,
+        getDataParamsFromAttributes(this.element));
+
+    /** @private {string} */
+    this.href_ = null;
+    const hrefWithVars = addParamsToUrl(this.shareEndpoint_, this.params_);
+    const urlReplacements = urlReplacementsFor(this.getWin());
+    urlReplacements.expand(hrefWithVars).then(href => {
+      this.href_ = href;
+    });
+
+    this.element.setAttribute('role', 'link');
+    this.element.addEventListener('click', () => this.handleClick_());
   }
 
-  /**
-   * Gets the configuration for the current social element
-   * @param {!element} element
-   * @return {?Object}
-   */
-  getElementConfig_() {
-    const script = elementByTag(this.element, 'script');
-    let config = {};
-    if (script) {
-      // Get config from script
-      try {
-        config = JSON.parse(script.textContent);
-      } catch (e) {
-        user.error(TAG, 'Malformed JSON configuration. %s', this.element);
-      }
-    } else {
-      // Get config from attributes
-      for (const param in this.typeConfig_['params']) {
-        if (this.element.hasAttribute('data-' + param)) {
-          config[param] = this.element.getAttribute('data-' + param);
-        }
-      }
-    }
-    return config;
+  /** @private */
+  handleClick_() {
+    const windowFeatures = 'resizable,scrollbars,width=640,height=480';
+    this.getWin().open(this.href_, '_blank', windowFeatures);
```

Comment:
You mean because urlReplacement might not have executed yet when the button is clicked?